### PR TITLE
Add type hints to arelle.XbrlUtil

### PR DESCRIPTION
**Pull request by selgamal**
#### Reason for change
Add type hints to arelle.XbrlUtil

#### Description of change
This pr includes 9 type ignores, all of them for `xmlValidate` function form `XmlValidate` module

#### Steps to Test

**review**:
@Arelle/arelle

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Security Insights
No security relevant content was detected by automated scans.

## Action Items
* Review PR for [security impact](https://wiki.atl.workiva.net/display/SECURITY/Development+Security+Review+Guidelines); comment "security review required" if needed or unsure
* Verify [`aviary.yaml`](https://wiki.atl.workiva.net/pages/viewpage.action?pageId=42762752#SecurityInsightFramework&Tooling-ConfiguringAviary) coverage of security relevant code
---
_Questions or Comments? Reach out on Slack: #support-infosec._

File: arelle/XbrlUtil.py [hunk 1]
```diff
@@ -49,14 +63,16 @@ def equalityHash(elt, equalMode=S_EQUAL, excludeIDs=NO_IDs_EXCLUDED):
                 return elt._hashXpathEqual
         except AttributeError:
             dts = elt.modelXbrl
+            from arelle.ModelXbrl import ModelXbrl
+            assert isinstance(dts, ModelXbrl), 'dts is not an instance of ModelXbrl'
             if not hasattr(elt,"xValid"):
-                xmlValidate(dts, elt)
-            hashableValue = elt.sValue if equalMode == S_EQUAL else elt.xValue
+                xmlValidate(dts, elt)  # type: ignore[no-untyped-call]
+            hashableValue = getattr(elt, 'sValue') if equalMode == S_EQUAL else getattr(elt, 'xValue')
```
Comment (sagesmith-wf) [inline arelle/XbrlUtil.py:70]: Why the change to `getattr`?

File: arelle/XbrlUtil.py [hunk 2]
```diff
@@ -93,21 +117,30 @@ def sEqual(dts1, elt1, elt2, equalMode=S_EQUAL, excludeIDs=NO_IDs_EXCLUDED, dts2
...
     return True
 
-def attributeDict(modelXbrl, elt, exclusions=set(), equalMode=S_EQUAL, excludeIDs=NO_IDs_EXCLUDED, ns2ns1Tbl=None, keyByTag=False, distinguishNaNs=False):
+def attributeDict(
+    modelXbrl: ModelXbrl,
+    elt: ModelObject,
+    exclusions: set[str] = set(),
+    equalMode: int = S_EQUAL,
+    excludeIDs: int = NO_IDs_EXCLUDED,
+    ns2ns1Tbl: dict[str, str] | None= None,
```
Comment (sagesmith-wf) [inline arelle/XbrlUtil.py:135]: Nit: Whitespace `None = None`

File: arelle/XbrlUtil.py [hunk 3]
```diff
@@ -135,45 +168,60 @@ def attributeDict(modelXbrl, elt, exclusions=set(), equalMode=S_EQUAL, excludeID
...
         return elt1.stringValue == elt2.stringValue
     elif equalMode == S_EQUAL: # formula WG e-mail 2018-09-06: or (equalMode == S_EQUAL2 and not isinstance(elt1.sValue, QName)):
         return elt1.sValue == elt2.sValue
     else: # includes dimension S-equal2, use xpath-2 equality.
-        if isinstance(elt1.xValue, DateTime) and isinstance(elt2.xValue, DateTime) and elt1.xValue.dateOnly != elt2.xValue.dateOnly:
+        elt1_xValue = elt1.xValue
+        elt2_xValue = elt2.xValue
+        if isinstance(elt1_xValue, DateTime) \
+            and isinstance(elt2_xValue, DateTime) \
+                and getattr(elt1_xValue, 'dateOnly') \
```
Comment (sagesmith-wf) [inline arelle/XbrlUtil.py:198]: This should be non-nullable even if its not now, its just a bool right?

File: arelle/XbrlUtil.py (hunk 1 above)
Comment (selgamal) [reply arelle/XbrlUtil.py:70]: That was before adding the `xValue` and `sValue` attrs to `ModelObject`, missed reverting this one

File: arelle/XbrlUtil.py (hunk 3 above)
Comment (selgamal) [reply arelle/XbrlUtil.py:198]: `dateOnly` gets assigned with __new__ of `DateTime`, mypy doesn't pick that up, maybe its better to add it as class attribute in #329, I will ingore[attr-defined] for now here.

File: arelle/XbrlUtil.py [hunk 4]
```diff
@@ -135,45 +168,59 @@ def attributeDict(modelXbrl, elt, exclusions=set(), equalMode=S_EQUAL, excludeID
...
     if not hasattr(elt2,"xValid"):
-        xmlValidate(elt2.modelXbrl, elt2)
+        xmlValidate(elt2.modelXbrl, elt2)  # type: ignore[no-untyped-call]
     if equalMode == VALIDATE_BY_STRING_VALUE:
         return elt1.stringValue == elt2.stringValue
     elif equalMode == S_EQUAL: # formula WG e-mail 2018-09-06: or (equalMode == S_EQUAL2 and not isinstance(elt1.sValue, QName)):
         return elt1.sValue == elt2.sValue
     else: # includes dimension S-equal2, use xpath-2 equality.
-        if isinstance(elt1.xValue, DateTime) and isinstance(elt2.xValue, DateTime) and elt1.xValue.dateOnly != elt2.xValue.dateOnly:
+        elt1_xValue = elt1.xValue
```
Comment (austinmatherne-wk) [inline arelle/XbrlUtil.py:194]: nit: use camel case to stay consistent with the rest of the arelle codebase

File: arelle/XbrlUtil.py [hunk 5]
```diff
@@ -135,45 +168,59 @@ def attributeDict(modelXbrl, elt, exclusions=set(), equalMode=S_EQUAL, excludeID
...
         if attrQname: # PSVI attribute value
-            modelAttribute = element.xAttributes[attrQname.clarkNotation]
+            modelAttribute = getattr(element, 'xAttributes')[attrQname.clarkNotation]
             if modelAttribute.xValid >= VALID:
                 return modelAttribute.xValue
         else: # PSVI element value (of text)
-            if element.xValid >= VALID:
-                return element.xValue
+            if getattr(element, 'xValid') >= VALID:
+                return getattr(element, 'xValue')
```
Comment (austinmatherne-wk) [inline arelle/XbrlUtil.py:221]: You don't need a getattr here after the ModelObject change

**Review by austinmatherne-wk — APPROVED** [review]

File: arelle/XbrlUtil.py (hunk 4 above)
Comment (selgamal) [reply arelle/XbrlUtil.py:194]: Actually the variables are not needed anymore, variables were just to make space because I was using `getattr`.

**Review by sagesmith-wf — APPROVED** [review]
